rs/ic_os/utils/src/deployment.rs

```diff
@@ -314,18 +366,8 @@ mod test {
     #[test]
     fn write_deployment() {
         let written_deployment =
-            serde_json::to_string_pretty::<DeploymentJson>(&DEPLOYMENT_STRUCT_NO_CPU).unwrap();
-
-        assert_eq!(DEPLOYMENT_STR_NO_CPU, written_deployment);
-
-        let written_cpu_deployment =
-            serde_json::to_string_pretty::<DeploymentJson>(&QEMU_CPU_DEPLOYMENT_STRUCT).unwrap();
-
-        assert_eq!(QEMU_CPU_DEPLOYMENT_STR, written_cpu_deployment);
-
-        let written_multi_url_deployment =
-            serde_json::to_string_pretty::<DeploymentJson>(&MULTI_URL_STRUCT).unwrap();
```

Comment:
I deleted these tests because they were breaking now that the Deployment struct has a new field—because when it would serialize, say, QEMU_CPU_DEPLOYMENT_STRUCT, it would add the mgmt_mac field as null, but the existing QEMU_CPU_DEPLOYMENT_STR wouldn't have that field.

I figured it was okay to delete these tests because we really only care about deserializing the json. We never serialize it back to json, so no need to test it.

Reply:
> I figured it was okay to delete these tests because we really only care about deserializing the json. We never serialize the deployment struct back to json, so no need to test it.

Do you mean that there's a lot of redundancy in tests because we are still keeping the one serialization test?

In general, I don't think it hurts to have some extra tests, especially since we may soon build on it with the rest of the config work.

If the only issue is that we're seeing the value serialize as `null`, I think we should change this behavior in the serializer instead. `#[serde(skip_serializing_if = "Option::is_none")]` is a pretty common pattern ([source](https://serde.rs/field-attrs.html#skip_serializing_if)).

Reply:
The way I understand it is the read_deployment test is the only one that really matters. read_deployment reads in the DEPLOYMENT_STR (deployment.json) and converts it to the DEPLOYMENT_STRUCT. Example:
```
let parsed_deployment: DeploymentJson = { serde_json::from_str(DEPLOYMENT_STR).unwrap() };

        assert_eq!(*DEPLOYMENT_STRUCT, parsed_deployment);
```

This is exactly what we do in production.

The write_deployment test converts the DEPLOYMENT_STRUCT back into a DEPLOYMENT_STR (deployment.json), but we never do this in production. We only deserialize, never serialize.

So when #[serde(skip_serializing_if = "Option::is_none")] is added and we serialize DEPLOYMENT_STRUCT in our test, it's actually not DEPLOYMENT_STR because DEPLOYMENT_STR has been updated to include mgmt_mac. Instead,  DEPLOYMENT_STRUCT serializes to _DEPLOYMENT_STR_NO_MGMT_MAC_. 

So we could make these changes to get the test to work. But like I said, read_deployment is the only test that really matters because we never serialize the deployment struct and I don't think we ever will. For this reason, I figured we could delete the test, but I left the single write_deployment assertion as a bit of a sanity check. Does that all make sense? WDYT?

Reply:
> This is exactly what we do in production.

I think it's OK to keep the "main" test in line with the current repo's config, but we also have all the state currently deployed to worry about. Checking that new code doesn't break with old values is exactly what these tests help us catch.

> But like I said, read_deployment is the only test that really matters because we never serialize the deployment struct and I don't think we ever will.

IMO more generally, tests like this give us the chance to check boundaries of code, even for cases that may not yet be fully exercised in reality. Though, I'm pretty sure we _are_ using serialization in [`setupos-inject-configuration`](https://github.com/dfinity/ic/blob/e93ea2e7492b6b66dcba3ab76fad1a168b234097/rs/ic_os/setupos-inject-configuration/src/main.rs#L253).

On `skip_serializing_if `, being that
```
{
    "key": "value",
    "key2": null
}
```
and
```
{
    "key": "value"
}
```
are equivalent as far as we're concerned, I tend to prefer the second, but really either one should be OK.

Reply:
Sounds good! I updated the test!